core/src/main/java/org/bitcoinj/crypto/AesKey.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright by the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bitcoinj.crypto;
+
+import org.bouncycastle.crypto.params.KeyParameter;
+
+/**
+ *  Wrapper for a byte array containing an AES Key. This is a replacement for Bouncy Castle's{@link KeyParameter} which
+ *  was used for this purpose in previous versions of bitcoinj. Unfortunately, this created a Gradle "api" dependency
+ *  on Bouncy Castle when that wasn't strictly necessary.
+ *  <p>
+ *  We have <i>tentatively</i> made this change without deprecation because it
+ *  affected so many method signatures and because updating is a trivial change. If for some
+ *  reason you have code that uses the Bouncy Castle {@link KeyParameter} type and need to convert
+ *  to {@code AesKey}, you can use {@link #ofKeyParameter(KeyParameter)} or {@link #toKeyParameter()}
+ *  // TODO: Update JavaDoc
+ */
+public class AesKey {
+    private final byte[]  key;
+
+    public AesKey(
+            byte[]  key)
+    {
+        this.key = key;
```

Comment:
Note that the replaced class is taking ownership of the key data (copying) here.

Nitpick: the formatting could be improved around this line.

Reply:
> Note that the replaced class is taking ownership of the key data (copying) here.
> 

My mistake. `byte[]` is mutable and should be defensively copied.

Reply:
Fixed.